### ISSUE TITLE
Ockam command man page generation

### ISSUE THREAD

**Issue by hairyhum** [build-trust/ockam#3808]
`ockam_command` is using `clap` for command line parsing. There is a tool to generate man files from `clap` config https://crates.io/crates/clap_mangen

With some improvements around https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/lib.rs to explain environment variables it can be used to produce decent enough documentation to existing commands.

**Comment by adrianbenavides** [comment]
I can take a look into this.

My main concern is about how to integrate it with our CI pipeline: maybe adding the man page generation in the `build.rs` will slow it down too much. And I never heard of the proposed alternative ([xtask](https://github.com/matklad/cargo-xtask)) Any preference/idea here @metaclips ?

**Comment by adrianbenavides** [comment]
I've done a bit of research on this. Here are the main findings:
- Trying to generate the man page from ockam_command's `build.rs` is not feasible as it needs to depend on itself to generate the docs. I've tried it and failed miserably
- The proper approach would be to have a separate package with the sole purpose of building the man page. The building process can then be automated with [cargo xtask](https://github.com/matklad/cargo-xtask). Here is [an example](https://github.com/ctrlcctrlv/humnumsort) of how to set this up

Other notes:
- `clap_mangen` is relatively new (February 2022)

### Generating manpages

I was only able to generate the manpage for the main command (the output of `ockam --help`). To generate manpages for subcommands, we should derive `Parser` instead of `Args` in every subcommand so we can run the following:

```rust
let cmd = <ockam_command::ProjectCommand as clap::CommandFactory>::command();
```

Some questions:
- Not sure this is the right way to do this. We need to figure out if `clap_manpage` can discover subcommands automatically
- How should the different manpages be organized?

**Comment by hargut** [comment]
Hi @hairyhum,
Hi @adrianbenavides,

I've looked into the topic and started to implement a subcommand that generates the man pages.

The PR adds a `manpages` subcommand, which when called, writes the man page files to one of the following directories:
- value from user provider `--dir|-d` option
- `~/.local/share/man/man1`
- current working directory `$PWD/ockam_man_pages` in case `$HOME` cannot be determined


Additionally I've added an option to disable gzip compression for the generated man pages. The current implementation generates one man page for the main command, and one per subcommand. Generation only done on users request calling the subcommand.

I'd be happy to receive your inputs and feedback.

Many thanks in advance.

Best regards,
Harald
